### ISSUE TITLE
Python 2 environment destroys python 3 conda installation

### ISSUE THREAD

**Issue by Zaharid** [conda/conda-env#79]
If I have a root conda environment working with python 3 and I do:

````shell
conda env update --name root --file environment.yml
````
Where environment.yml contains a Python 2 environment, ie

````shell
$ grep python environment.yml
- python=2.7.9=2
````

Then everything gets completely destroyed, for example:

```shell
$ conda
bash: /home/zah/anaconda3/bin/conda: /home/zah/anaconda3/bin/python3: bad interpreter: No such file or directory
$ ipython
bash: /home/zah/anaconda3/bin/ipython: /home/zah/anaconda3/bin/python3: bad interpreter: No such file or directory
$ python
Python 2.7.9 |Continuum Analytics, Inc.| (default, Mar  9 2015, 16:20:48) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://binstar.org
>>> 
````

**Comment by tswicegood** [comment]
This is a broader issue with conda.  For example, if you had python 3 installed then did:

```
conda install python=2.7.9
```

You would end up in the same state.  I agree that this is a problem (see https://github.com/conda/conda/issues/1268 for one of the latest bug reports on this), but I think it should be handled at the conda level, not conda-env.  Otherwise, this package will handle installation different than conda does.

I'm going to leave this open in the meantime just to track that it's here.  Once conda's behavior changes then I'll mark this as complete.

cc @ilanschnell @asmeurer

**Comment by asmeurer** [comment]
conda-env is doing something less than conda here, though. If you change Python versions in the root environment with `conda install` then conda itself will still work, because conda always includes itself in the root environment specs.  

Everything else will break though because the Python 2 versions of the packages will still be installed. https://github.com/conda/conda/issues/454 is sort of the real issue here. Although it would not be hard to make a command that works like `conda update --all` (i.e., "installs" everything into the environment) but that also lets you change the Python version. In fact, perhaps `conda update --all` should just allow additional specs to be included, so that `conda update --all python=3.4` would update a Python 2 environment to Python 3 (or fail if it isn't possible). 

Anyway, I thought the whole point of conda-env was that the environment is named in the environment.yml. Why does it even allow -n? At the very least, maybe `conda env -n root` should not work.

**Comment by malev** [comment]
You can't name a new environment with the same name as the root environment.
```
➜  tmp  conda env create -n root
Error: prefix already exists: /Users/mvanetta/miniconda3
```
